src/VisualStudio/Core/Def/Venus/ContainedLanguage.cs

```diff
@@ -115,13 +96,19 @@ public static string GetFilePathFromHierarchyAndItemId(IVsHierarchy hierarchy, u
             var bufferTagAggregatorFactory = ComponentModel.GetService<IBufferTagAggregatorFactoryService>();
             _bufferTagAggregator = bufferTagAggregatorFactory.CreateTagAggregator<ITag>(SubjectBuffer);
 
+            // TODO: Can contained documents be linked or shared?
+            this.DataBuffer.Changed += OnDataBufferChanged;
```

Comment:
Was there a reason to move this earlier? There's a potential risk here that if that event can fire while we're initializing this type that the event can fire while the type isn't fully constructed yet.

Reply:
Oops I moved this when I was initially splitting the constructors and forgot to put it back.